pkg/git/libgit2/managed_test.go

```diff
@@ -552,6 +552,31 @@ func TestManagedCheckoutBranch_Checkout(t *testing.T) {
 	g.Expect(err).ToNot(HaveOccurred())
 	g.Expect(cc.String()).To(Equal(git.DefaultBranch + "/" + commit.Id().String()))
 	g.Expect(git.IsConcreteCommit(*cc)).To(Equal(true))
+
+	// Create a new branch and push it.
+	err = createBranch(repo, "test", nil)
+	g.Expect(err).ToNot(HaveOccurred())
+	transportOptsURL := getTransportOptionsURL(git.HTTP)
+	managed.AddTransportOptions(transportOptsURL, managed.TransportOptions{
+		TargetURL: repoURL,
+	})
+	defer managed.RemoveTransportOptions(transportOptsURL)
+	origin, err := repo.Remotes.Create("origin", transportOptsURL)
+	g.Expect(err).ToNot(HaveOccurred())
+	err = origin.Push([]string{"refs/heads/test:refs/heads/test"}, &git2go.PushOptions{})
+	g.Expect(err).ToNot(HaveOccurred())
+
+	branch.Branch = "test"
+	tmpDir2 := t.TempDir()
+	cc, err = branch.Checkout(ctx, tmpDir2, repoURL, authOpts)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// Check if the repo HEAD points to the branch.
+	repo, err = git2go.OpenRepository(tmpDir2)
+	g.Expect(err).ToNot(HaveOccurred())
+	head, err := repo.Head()
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(head.Branch().Name()).To(Equal("test"))
```

Comment:
I've noticed that some of the above and some other new tests don't call free. It may not matter much in the tests but it'd be good to have good practices of using git2go in the tests as well.

Reply:
oops, missed those, thanks for pointing out